client/src/components/layouts/default.tsx

```diff
@@ -134,6 +140,18 @@ function DefaultLayout({
   updateAllChallengesInfo
 }: DefaultLayoutProps): JSX.Element {
   const { t } = useTranslation();
+  const isMobileLayout = useMediaQuery({ maxWidth: MAX_MOBILE_WIDTH });
+  const isMultifileEditorChallenge =
+    superBlock === SuperBlocks.RespWebDesignNew ||
+    superBlock === SuperBlocks.JsAlgoDataStructNew ||
```

Comment:
I know there is a similar logic used for multifileCertProjects; however, this would break for new superblocks. 
It would be great if you could explore a more rigid solution such as using the usesMultifileEditor prop.

Reply:
> It would be great if you could explore a more rigid solution such as using the usesMultifileEditor prop.

I definitely agree with you so I will try to push the code with using the prop.